docs/naps/8-telemetry.md

```diff
@@ -0,0 +1,182 @@
+
+In the main package, there is a need to add code to ask users if they want to enable telemetry. This code should be executed only once per environment.
+
+Telemetry should contain following ways to disable it:
+
+1. Disable in settings
+2. uninstall `napari-telemetry` package
+3. Environment variable `NAPARI_TELEMETRY=0`
+4. Full list of endpoints used for collecting telemetry, that could be filtered on the firewall level.
+
+The user should be able to adjust the telemetry level of detail. The following levels are proposed:
+
+1. `none` - no telemetry is collected
+2. `basic` - information about the napari version, python version, OS, and CPU architecture is collected and if it is the first report by the user. There is also a user identifier created based on computer details that will be rerendered each week to prevent tracking the user, but allow to not count a user multiple times. 
+3. `middle` - same as in `basic` but also information about the list of installed plugins and their versions is collected. We take care to not collect data about plugins that are not intended to be public. 
+4. `full` - same as in `middle` but also collects information about plugin usage by binding to app-model and logging plugin commands used. Also basic information about data like type (`np.ndarray`, `dask.array`, `zarr.Array`, etc.) and its size is collected.
+
+There should be a visible indicator that telemetry is enabled (for example on the status bar). 
+
+The second part of this work should be to setup the server to collect telemetry data. After collecting data, it should provide a basic public dashboard that will allow the community to see aggregated information.
+
+I propose to have the following data retention policy:
+
+1) Up to 2 weeks for logs.
+2) up 2 months of raw data (1 month of collection, then aggregation and time to validate aggregated data),
+3) infinite of aggregated data.
+  
+## Privacy assessment
+
+During the preparation of this NAP we assume that none of the collected data will be presented in 
+a form that allows to identify a single user or identify a research area of user. We also select a set of data that will be collected to minimize the possibility of revealing fragile data, but it is impossible to guarantee that it will not be possible to identify a single user (for example by checking installed plugin combinations).
+
+Because of this, we propose to not publish raw data and only show aggregated results. The aggregation will be performed using scripts. Napari core devs will access raw data only if there are errors in the aggregation process.
+
+We also will publish a list of endpoints for each level of telemetry, so the given level of telemetry could be blocked on the organization level (for example by the rule on the firewall).
```

Comment:
For everything that is plugin-related, you likely want to filter-out everything that is outside of a explicit-exclusion list. Local plugin names may leak internal information.

Reply:
I have clarified that plugin need to be uploaded to PyPI with the napari classifier (to be searchable in the napari plugin manager) and be in the stable version.